render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
+	// Present the object.
+	if presenter, ok := ctx.Value(presenterCtxKey).(Presenter); ok {
+		v = presenter.Present(ctx, v)
+	} else {
+		v = DefaultPresenter.Present(ctx, v)
+	}
+
+	// Format data based on Content-Type.
+	switch contentType, _ := ctx.Value(contentTypeCtxKey).(ContentType); contentType {
+	case ContentTypeJSON:
+		JSON(ctx, w, v)
+	case ContentTypeXML:
+		XML(ctx, w, v)
+	default:
+		JSON(ctx, w, v)
+	}
```

Comment:
yea, the interface for those methods, HTML and PlainText/String would have to be specific to the kind of data they want to receive